### ISSUE TITLE
Array of embeddable with date property always updated

### ISSUE THREAD

**Issue by vinverdy** [mikro-orm/mikro-orm#4360]
**Describe the bug**
always updated the row, maybe related with https://github.com/mikro-orm/mikro-orm/issues/4078

```ts
/* istanbul ignore file */
/* eslint-disable */
import { Embeddable, Embedded, Entity, PrimaryKey, Property } from '@mikro-orm/core';
import { MikroORM } from '@mikro-orm/postgresql';

@Embeddable()
export abstract class Animal {
  @Property()
  public readonly birthday!: Date;
}

@Entity()
export class Owner {
  @PrimaryKey()
  id!: number;

  @Embedded(() => Animal, { array: true })
  pet!: Animal[];
}

let orm: MikroORM;

beforeAll(async () => {
  orm = await MikroORM.init({
    entities: [Owner],
    debug: true,
    allowGlobalContext: true,
  });

  await orm.schema.refreshDatabase();
});

afterAll(async () => {
  await orm.close(true);
});

it('doing some update when shouldnt', async () => {
  await orm.em.insert(Owner, {
    id: 1,
    pet: [
      {
        birthday: new Date(),
      },
    ],
  });
  await orm.em.findOneOrFail(Owner, 1);
  await orm.em.flush();
});
```
![image](https://github.com/mikro-orm/mikro-orm/assets/37362021/bb37b685-2664-4a2e-ac7b-cc1a84f5729b)

**Comment by B4nan** [comment]
The solution to this will be the same as for #4362, which is enforcing the conversion via the custom type instead of relying on the driver. Will close this as a duplicate, please subscribe there.

As a workaround, you can specify the custom type explicitly:

```ts
@Property({ type: DateTimeType })
birthday!: Date;
```

**Comment by B4nan** [comment]
This was actually a bit different issue that I originally though, but it will be also resolved by #4391.
